cs-amazon/src/main/java/io/cloudslang/content/amazon/factory/InputsWrapperBuilder.java

```diff
@@ -69,10 +77,13 @@ private InputsWrapperBuilder() {
         return wrapper;
     }
 
-    private static HttpClientInputs getHttpClientInputs(CommonInputs commonInputs) {
+    @SafeVarargs
+    private static <T> HttpClientInputs getHttpClientInputs(CommonInputs commonInputs, T... builders) {
         HttpClientInputs httpClientInputs = new HttpClientInputs();
 
-        httpClientInputs.setUrl(InputsUtil.getUrlFromApiService(commonInputs.getEndpoint(), commonInputs.getApiService()));
+        String prefix = getPrefix(builders);
+
+        httpClientInputs.setUrl(InputsUtil.getUrlFromApiService(commonInputs.getEndpoint(), commonInputs.getApiService(), prefix));
```

Comment:
@Tirla-Alin Endpoint is not the same because of the different API's (EC2 and S3). For S3 the calls can have different endpoints depends an what call is made. If you have a better way to do it... I'm open to different approaches...